pkg/workflows/management/gitops.go

```diff
@@ -0,0 +1,91 @@
+package management
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type pauseGitOpsReconcile struct{}
+
+// Run pauseGitOpsReconcile pause GitOps reconciler before management cluster upgrade.
+func (s *pauseGitOpsReconcile) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Pausing GitOps cluster resources reconcile")
+	err := commandContext.GitOpsManager.PauseClusterResourcesReconcile(ctx, commandContext.ManagementCluster, commandContext.ClusterSpec, commandContext.Provider)
```

Comment:
Should we use current cluster spec here instead so it tries to pause the current objects and not any new ones?
```suggestion
	err := commandContext.GitOpsManager.PauseClusterResourcesReconcile(ctx, commandContext.ManagementCluster, commandContext.CurrentClusterSpec, commandContext.Provider)
```

Reply:
Discussed this offline. `ClusterSpec` works with current flow.